llama-index-integrations/vector_stores/llama-index-vector-stores-elasticsearch/llama_index/vector_stores/elasticsearch/base.py

```diff
@@ -239,7 +239,7 @@ async def _create_index_if_not_exists(
             index_name: Name of the AsyncElasticsearch index to create.
             dims_length: Length of the embedding vectors.
         """
-        if self.client.indices.exists(index=index_name):
+        if await self.client.indices.exists(index=index_name):
```

Comment:
@logan-markewich IIRC I think there was some reason as to why we removed the `await`? Or maybe I'm wrong and it never was there to begin with.

I do think we need to have this since the client is and async client though.

Reply:
for visibility on this issue:

- it appears a lower version of `elasticsearch` python client actually does not yield a Coroutine when `clinet.indices.exists` was invoked.
- this is no longer the case for the most recent version of `elasticsearch` (at time of this comment, v8.12.1).
- since v8.12.1 is the minimum requirement for this integration (i.e. see pyproject.toml), we should be safe to add back `await` now.